controllers/algae_measurement.py

```diff
@@ -0,0 +1,45 @@
+import math
+
+from magicbot import StateMachine, state
+
+from components.algae_manipulator import AlgaeManipulatorComponent
+
+
+class AlgaeMeasurement(StateMachine):
+    algae_manipulator_component: AlgaeManipulatorComponent
+
+    def __init__(self) -> None:
+        self.stopped_counter = 0
+        self.starting_position = 0.0
+
+    @state(
+        first=True,
+        must_finish=True,
+    )
+    def measuring(self, initial_call) -> None:
+        if initial_call:
+            self.stopped_counter = 0
+            self.starting_position = (
+                self.algae_manipulator_component.get_injector_position()
+            )
+
+        self.algae_manipulator_component.measure_algae()
+
+        if math.isclose(
+            self.algae_manipulator_component.get_injector_velocity(), 0.0, abs_tol=0.1
+        ):
+            self.stopped_counter += 1
+
+        else:
+            self.stopped_counter = 0
+
+        if (
+            self.algae_manipulator_component.bottom_flywheel_speed()
+            or self.algae_manipulator_component.top_flywheel_speed() >= 0.25
```

Comment:
You need to do the comparison on both flywheels .
```suggestion
            self.algae_manipulator_component.bottom_flywheel_speed() >= 0.25
            or self.algae_manipulator_component.top_flywheel_speed() >= 0.25
```